dlrs/main/classes/RollupSummariesSelectorTest.cls

```diff
@@ -0,0 +1,589 @@
+@isTest
+private with sharing class RollupSummariesSelectorTest {
```

Comment:
The test class already creates a MockedCustomMetadataSelector. Instead of testing only the CustomMetadataSelector, we could pass the mocked version to a RollupSummariesSelector instance through a @TestVisible constructor and test the outer class itself.

Example snippet Andy wrote:

 ```
 public RollupSummariesSelector(boolean enforceSecurity, boolean forEdit) {
    // Dual selectors for querying custom object and custom metadata stores
    this(new CustomObjectSelector(enforceSecurity, forEdit), 
         new CustomMetadataSelector(enforceSecurity,forEdit));
  }

  @TestVisible
  private RollupSummariesSelector(boolean enforceSecurity, boolean forEdit, CustomObjectSelector customObjectSelector, CustomMetadataSelector customMetadataSelector) {
    // Dual selectors for querying 
  private static void selectAll_notForEdit_MDT() {

    // Given
    MockedCustomMetadataSelector mock = new MockedCustomMetadataSelector(false);
    
    // When
    Test.startTest();
    RollupSummariesSelector rollupSelector =
        new RollupSummariesSelector(false, false, 
          new RollupSummariesSelector.CustomObjectSelector(),
          mock);
    List<RollupSummary> summaries = rollupSelector.selectAll();
    Test.stopTest();

    // Then
```